src/cryptography/hazmat/backends/openssl/backend.py

```diff
@@ -136,6 +138,44 @@ def _encode_basic_constraints(backend, basic_constraints):
     return pp, r
 
 
+def _encode_subject_alt_name(backend, san):
+    general_names = backend._lib.GENERAL_NAMES_new()
+    assert general_names != backend._ffi.NULL
+    general_names = backend._ffi.gc(
+        general_names, backend._lib.GENERAL_NAMES_free
+    )
+
+    for alt_name in san:
+        gn = backend._lib.GENERAL_NAME_new()
```

Comment:
ffff. thanks

On Sun, Jul 5, 2015 at 12:32 PM, Paul Kehrer <notifications@github.com>
wrote:

> In src/cryptography/hazmat/backends/openssl/backend.py
> <https://github.com/pyca/cryptography/pull/2085#discussion_r33896254>:
>
> > @@ -136,6 +138,44 @@ def _encode_basic_constraints(backend, basic_constraints):
> >      return pp, r
> >
> >
> > +def _encode_subject_alt_name(backend, san):
> > +    general_names = backend._lib.GENERAL_NAMES_new()
> > +    assert general_names != backend._ffi.NULL
> > +    general_names = backend._ffi.gc(
> > +        general_names, backend._lib.GENERAL_NAMES_free
> > +    )
> > +
> > +    for alt_name in san:
> > +        gn = backend._lib.GENERAL_NAME_new()
>
> This will leak if the general name is not a DNSName right now.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/pyca/cryptography/pull/2085/files#r33896254>.
>



-- 
"I disapprove of what you say, but I will defend to the death your right to
say it." -- Evelyn Beatrice Hall (summarizing Voltaire)
"The people's good is the highest law." -- Cicero
GPG Key fingerprint: 125F 5C67 DFE9 4084